server/api/users/posts/[id].put.ts

```diff
@@ -0,0 +1,81 @@
+import { Posts, Users } from "../../../dbModels";
+import middlewareFunction from "../../../utils/middlewareFunction";
+
+interface IRequestBody {
+	channelData: string;
+}
+
+export default defineEventHandler(async (e) => {
+	//IF THE USER HAS EDIT OR ALL PERMISSIONS TO THIS CHANNEL THEN THE USER CAN EDIT ITS POSTS
+	const postId = e.context.params.id;
+	console.log(`PUT /api/user/posts/updatePost/${postId}`);
+	try {
+		const userId = await middlewareFunction(e);
+		console.log(userId);
+
+		if (e.req.headers.authentication == null) {

```

Comment:
https://v3.nuxtjs.org/getting-started/error-handling/

https://github.com/nuxt/framework/discussions/1485

nuxt internally h3 which gives more easier way to handle req and response events.Please read about it

Reply:
Sure, I will give it a read and see what can be done.